osu.Game/Screens/Play/Player.cs

```diff
@@ -222,6 +222,12 @@ public void TogglePaused()
             if (IsPaused) Pause(); else Resume();
         }
 
+        protected override void Dispose(bool isDisposing)

```

Comment:
You're right. But in removing this I noticed that player is currently disposed one Screen.Exit too late (requires exiting back to main menu) so I'll address that too.